rfcs/62-editor-data-model.md

```diff
@@ -0,0 +1,562 @@
+# Feature Name: `editor_data_model`
+
+## Summary
+
+This RFC describes the data model and API that the Editor uses in memory, and the basis for communicating with the Game process and for serializing that data to disk (although the serialization format to disk is outside of the scope of this RFC). The data model is largely inspired by the one of [Our Machinery](https://ourmachinery.com/), as described in [this blog post](https://ourmachinery.com/post/the-story-behind-the-truth-designing-a-data-model/).
+
+## Glossary
+
+In this RFC, the following terms are used:
+
+- **Game** refers to any application made with Bevy, which encapsulates proper games but also any other kind of Bevy application (CAD, architecture software, _etc._).
+- **Author** refers to any person (including non-technical) building the Game with the intent of shipping it to end users.
+- **Editor** refers to the software used to edit Game data and prepare them for runtime consumption, and which this RFC describes the data model of.
+- **Runtime** refers to the Game or Editor process running time, as opposed to compiling or building time. When not otherwise specified, this refers to the Game's runtime. In general it's synonymouns of a process running, and is opposed to _build time_ or _compile time_.

```

Comment:
> In general it's synonymouns of a process running,

Here we define “Runtime” as both the time window where the Game or Editor process runs, and the running process itself.

From what I've read in this RFC, it looks like that “Runtime” is just used in reference to a running Game, to distinguish it from its source or executable. This term is also never used in reference to the Editor, which is not needed, since the Editor is always running (hopefully no crashes 😰) during the Author's working session. Therefore I propose to simplify this definition to something like: “**Runtime** refers to the running Game process.”.

Reply:
This was fixed already in a previous change.